af_ktls.c

```diff
@@ -984,10 +988,14 @@ static inline ssize_t tls_peek_data(struct tls_sock *tsk, unsigned flags)
 			goto peek_failure;
 		}
 
+		if (peeked_size < KTLS_MIN_RECORD_SIZE(tsk)) {
+			ret = -EBADF;
```

Comment:
I am discussing this because of design so we can adopt return values to user space.

So to be clear:
* for DTLS/UDP:
   * EBADF when peeked size is bigger then we expect based on header
   * EBADF when peeked size is smaller then we expect based on header

* for TLS/TCP:
  * when peeked size is bigger than size based on TLS header, process only TLS record - part based on TLS header
  * when peeked size is smaller than size based on TLS header or it does not cover header at all, we should signalize it to user space, shouldn't we?